### ISSUE TITLE
[a11y]: Datatable [with-radio-selection] - Unable to select item using keyboard(Enter key)

### ISSUE THREAD

**Issue by diksha-22-crypto** [carbon-design-system/carbon#18856]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v11

### React version

v17.0.2

### Automated testing tool and ruleset

manual

### Assistive technology

No

### Description

Issue: While navigating using keyboard(Tab) in a data table having radio buttons, user is not able to make the selection using 'ENTER' key. Key UP and DOWN works when there are more than 1 record in the table but for single record, selection is blocked.
The issue also persists when radioButton is used standalone(not inside the data table).

Expected: User should be able to select the record using keyboard.

### Reproduction/example

Video of the issue is attached in STR section.

### Steps to reproduce

https://github.com/user-attachments/assets/4de064bd-addc-4141-90e9-546d5d4a5225

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by alisonjoseph** [comment]
Hi there, thanks for opening this issue. Testing within storybook and I'm able to select using the arrow keys or space bar which matches up with the ARIA spec. https://www.w3.org/WAI/ARIA/apg/patterns/radio/

**Comment by diksha-22-crypto** [comment]
Thanks @alisonjoseph for the response. I just tested my application and the space bar is working for me as well when there is only 1 record to select. Though arrow keys works only when there are more than 1 records.
Since it is working fine hence closing the issue!
